### DISCUSSION TITLE
What is the syntax of the "Matching pattern" field for ruleset metadata restrictions?

### DISCUSSION THREAD

**Discussion by justinmchase** [community/community#61655] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I would like to add a metadata restriction for email addresses to come from my own company, but I need to make some exceptions for some notable bots.

So in this UI it says "Matching pattern"

![image](https://github.com/community/community/assets/10974/78bc52f9-dd27-4de5-93b7-7a01b7ae249e)

But what is the syntax for the pattern? Is it glob or regexp or something else? How can I essentialy express a logical OR between acceptable email domains?

For example I want it to essentially have this regexp logic:

```
/(example\.com|\[bot\]@users\.noreply\.github\.com)$/
```

**Comment by IanTaylor-catapult** [comment]
I would also like to know this. Looking to enforce commit message formats if possible.

**Comment by johnecon** [comment, marked as answer]
check here: https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#enforce-conventional-commits
